pkg/providers/amifamily/ami_test.go

```diff
@@ -12,178 +12,346 @@ See the License for the specific language governing permissions and
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = Describe("AMI Provider", func() {
+	var version string
+	BeforeEach(func() {
+		version = lo.Must(awsEnv.AMIProvider.KubeServerVersion(ctx))
 	})
-	It("should allow prefixed id, prefixed name, and prefixed owners", func() {
-		amiSelector := map[string]string{
-			"aws::name":   "my-ami",
-			"aws::ids":    "ami-abcd1234,ami-cafeaced",
-			"aws::owners": "self,amazon",
-		}
-		filters, owners := getFiltersAndOwners(amiSelector)
-		Expect(owners).Should(ConsistOf(defaultOwners))
-		Expect(filters).Should(ConsistOf([]*ec2.Filter{
-			{
-				Name:   aws.String("name"),
-				Values: aws.StringSlice([]string{"my-ami"}),
-			},
-			{
-				Name: aws.String("image-id"),
-				Values: aws.StringSlice([]string{
-					"ami-abcd1234",
-					"ami-cafeaced",
-				}),
+	It("should succeed to resolve AMIs (AL2)", func() {
+		nodeTemplate = test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				AMIFamily: &v1alpha1.AMIFamilyAL2,
 			},
-		}))
+		})
+		awsEnv.SSMAPI.Parameters = map[string]string{
+			fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id", version):       amd64AMIName,
+			fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2-gpu/recommended/image_id", version):   amd64NvidiaAMIName,
+			fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2-arm64/recommended/image_id", version): arm64AMIName,
+		}
+		amis, err := awsEnv.AMIProvider.Get(ctx, nodeTemplate, &amifamily.Options{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(amis).To(HaveLen(4))
 	})
-	It("should allow prefixed name and prefixed owners", func() {
-		amiSelector := map[string]string{
-			"aws::name":   "my-ami",
-			"aws::owners": "0123456789,self",
+	It("should succeed to resolve AMIs (Bottlerocket)", func() {
+		nodeTemplate = test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				AMIFamily: &v1alpha1.AMIFamilyBottlerocket,
+			},
+		})
+		awsEnv.SSMAPI.Parameters = map[string]string{
+			fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/x86_64/latest/image_id", version):        amd64AMIName,
+			fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s-nvidia/x86_64/latest/image_id", version): amd64NvidiaAMIName,
+			fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/arm64/latest/image_id", version):         arm64AMIName,
+			fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s-nvidia/arm64/latest/image_id", version):  arm64NvidiaAMIName,
 		}
-		filters, owners := getFiltersAndOwners(amiSelector)
-		Expect(owners).Should(ConsistOf([]*string{
-			aws.String("0123456789"),
-			aws.String("self"),
-		}))
-		Expect(filters).Should(ConsistOf([]*ec2.Filter{
-			{
-				Name:   aws.String("name"),
-				Values: aws.StringSlice([]string{"my-ami"}),
+		amis, err := awsEnv.AMIProvider.Get(ctx, nodeTemplate, &amifamily.Options{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(amis).To(HaveLen(6))
```

Comment:
Same here for 6, shouldn't it be 4?

Reply:
Same response as above. See the requirements sets. We have to create disparate sets of the requirements that can be ORed together